### PR TITLE
fix(packager): support node-haste browser and react-native replacements on Windows

### LINKED ISSUE #9570: NPM browser field not picked up on packager server running on Windows
The packager server uses browser versions of an NPM package when available (e.g., for uuid).  On Windows, this replacement behavior is not working correctly.  Here's a simple repro:
```
var uuid = require('uuid');
```
Do this in anywhere in your react-native app and it will fail to pick up the browser version of that package.

This is on Windows 10, react-native@0.31.0, node@6.2.1.  Will confirm if still exists in 0.32.0-rc.0.

### PR DESCRIPTION
**Pull request by rozele**
Explain the **motivation** for making this change. What existing problem does the pull request solve?

Fix issue with browser and react-native module mappings in node-haste when running the packager from Windows.

**Test plan (required)**

Tested on the NPM package uuid, which has a browser mapping, and it worked.

Fixes #9570

### CODE REVIEW

**Comment by facebook-github-bot-2** [comment]
By analyzing the blame information on this pull request, we identified @davidaurelio to be a potential reviewer.

**Comment by satya164** [comment]
cc @davidaurelio

File: packager/react-packager/src/node-haste/Package.js [hunk 1]
```diff
@@ -58,19 +58,19 @@ class Package {
...
         return replacement === false
           ? false
           : replacement || name;
       }
 
-      if (!isAbsolutePath(name)) {
-        throw new Error(`Expected ${name} to be absolute path`);
+      const relPath = './' + path.relative(this.root, name);
+      if (path.sep !== '/') {
+        relPath.replace(path.sep, '/');
```
Comment (rozele) [inline packager/react-packager/src/node-haste/Package.js]: probably should assign this...

**Comment by facebook-github-bot-7** [comment]
@rozele updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9572/files/ecfcf534a395fbde50ba33cae9fe7f67419141c3..976d61caf137470c37a6fd0766fac173a3495f7a)

**Comment by rigdern** [comment]
How is the code here kept in sync with the code in the node-haste repo? Package.js in the node-haste repo has the same bug: https://github.com/facebook/node-haste/blob/58bc0782858098c57a8a61f387b2fb5852c39550/src/Package.js#L61

**Comment by mkonicek** [comment]
Pinged @davidaurelio to take a look.

**Comment by mkonicek** [comment]
Could you please add a short comment in the code explaining why the `redirectRequire` function exists and what it does?

I find it scary there's a `redirectRequire` function in quite a core file that has to do with how requires work and there's almost no documentation in that file :( (I know this was the state before this PR :))

Thanks for the fix! 👍 I wanted to merge it now to unblock but have so little context on what this does we should get someone who understands the codebase to look.

**Comment by davidaurelio** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-5** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://phabricator.intern.facebook.com/D3821761) to review internal test results.

**Pull request closed without merging**